src/refiners/training_utils/trainer.py

```diff
@@ -346,18 +346,19 @@ def backward(self) -> None:
         self._call_callbacks(event_name="on_backward_begin")
         scaled_loss = self.loss / self.clock.num_step_per_iteration
         backward(tensors=scaled_loss)
-        self._call_callbacks(event_name="on_backward_end")
-        if self.clock.is_optimizer_step:
-            self._call_callbacks(event_name="on_optimizer_step_begin")
-            self.optimizer.step()
-            self.optimizer.zero_grad()
-            self._call_callbacks(event_name="on_optimizer_step_end")
-        if self.clock.is_lr_scheduler_step:
-            self._call_callbacks(event_name="on_lr_scheduler_step_begin")
-            self.lr_scheduler.step()
-            self._call_callbacks(event_name="on_lr_scheduler_step_end")
-        if self.clock.is_evaluation_step:
-            self.evaluate()
+        if self.global_step % self.clock.num_step_per_iteration == 0:
```

Comment:
Isn't that what self.clock.is_optimizer_step is checking?

Reply:
ah, you are right. I think this pr is already implemented then. Closing now